packages/core/src/agent/MessageReceiver.ts

```diff
@@ -9,21 +9,29 @@ import { Lifecycle, scoped } from 'tsyringe'
 
 import { AriesFrameworkError } from '../error'
 import { ConnectionService } from '../modules/connections/services/ConnectionService'
+import { ProblemReportError, ProblemReportMessage } from '../modules/problem-reports'
 import { JsonTransformer } from '../utils/JsonTransformer'
 import { MessageValidator } from '../utils/MessageValidator'
 import { replaceLegacyDidSovPrefixOnMessage } from '../utils/messageType'
 
+import { CommonMessageType } from './../modules/common/messages/CommonMessageType'
 import { AgentConfig } from './AgentConfig'
 import { Dispatcher } from './Dispatcher'
 import { EnvelopeService } from './EnvelopeService'
+import { MessageSender } from './MessageSender'
 import { TransportService } from './TransportService'
+import { createOutboundMessage } from './helpers'
 import { InboundMessageContext } from './models/InboundMessageContext'
 
+export enum ProblemReportReason {
+  MessageParseFailure = 'message-parse-failure',
```

Comment:
Just from my curiosity... Is `message-parse-failure` part of some RFC?

Maybe, we can move it to `modules/problem-reports`. What do you think?

Reply:
I have referred to the Aries cloud agent python codebase and they are using the same enum https://github.com/hyperledger/aries-cloudagent-python/blob/main/aries_cloudagent/core/dispatcher.py#L150

Reply:
Yes, We can move it to modules/problem-reports